### PR TITLE
Bug2956 and dftfilter bugfix

### PR DESCRIPTION
**Pull request by sabineleske**
These commits add the method "spectrum interpolation" as a new option to reduce power line noise in data, as described in: http://bugzilla.fieldtriptoolbox.org/show_bug.cgi?id=2956 
New code and documentation was added to the functions: 
private/preproc.m 
 preproc/ft_preproc_dftfilter.m
Documentation was added in: 
ft_preprocessing.m
New function to demonstrate the functionality was added here: test/test_bug2956.m

The last commit contains a bugfix for ft_preproc_dftfilter.m. The problem related to this bug is described here:
http://bugzilla.fieldtriptoolbox.org/show_bug.cgi?id=3216

### CODE REVIEW

File: preproc/ft_preproc_dftfilter.m [hunk 1]
```diff
@@ -1,27 +1,55 @@
-function [filt] = ft_preproc_dftfilter(dat, Fs, Fl)
+function [filt] = ft_preproc_dftfilter(dat, Fs, Fl, Flreplace, Flwidth, Neighwidth)
```
Comment (schoffelen) [inline preproc/ft_preproc_dftfilter.m]: for  backward compatibility: could you make the new additional input arguments optional, e.g. 

ft_preproc_dftfilter(dat, Fs, Fl, varargin)

if numel(varargin)>0
  Flreplace = varargin{1};
  Flwidth etc
else 
  Flreplace = 'zero';
end

**Comment by schoffelen** [comment]
What's the status of this?

**Comment by sabineleske** [comment]
Sorry for the delay, the cfg options are already changed as optional. But there was a new idea for an additional optional argument (related to the data length/ SR for spectrum interpolation), that needs to be tested first. The commits will be submitted in the next days.

**Comment by sabineleske** [comment]
The new commits contain the following changes:
- new cfg input argument are optional
- small fix in preproc/ft_preproc_dftfilter.m
- additional documentation on how to apply spectrum interpolation for short trials
- error message in case padding by data mirroring is requested, because it is not applicable for spectrum interpolation

**Pull request merged**